assets/src/blocks/ShareButtons/ShareButtonsBlock.js

```diff
@@ -0,0 +1,87 @@
+import { ShareButtonsEditor } from './ShareButtonsEditor';
+import { ShareButtonsFrontend } from './ShareButtonsFrontend';
+
+const { registerBlockType } = wp.blocks;
+const { __ } = wp.i18n;
+
+const BLOCK_NAME = 'planet4-blocks/share-buttons';
+
+export const registerBlock = () => {
+  return registerBlockType(BLOCK_NAME, {
+    title: 'Share Buttons (beta)',
+    icon: 'feedback',
+    category: 'planet4-blocks',
+    supports: {
+      html: false,
+    },
+    attributes: {
+      url: {
+        type: 'string',
+        default: document.URL,
+      },
+      openInNewTab: {
+        type: 'boolean',
+        default: true,
+      },
+      gaCategory: {
+        type: 'string',
+      },
+      gaAction: {
+        type: 'string',
+      },
+      gaLabel: {
+        type: 'string',
+        default: 'n/a',
+      },
+      utmMedium: {
+        type: 'string',
+      },
+      utmContent: {
+        type: 'string',
+      },
+      utmCampaign: {
+        type: 'string',
+      },
+      buttons: {
+        type: 'object',
+        default: {
+          facebook: {
```

Comment:
Is there a reason for using an object over an array, for these buttons?

Reply:
Yes, It's more performant than using the `filter` for Arrays. In this case you will only update a `key|value` instead of finding an object, update and replace it into the Array

Reply:
Can you give me an indication about how much performance difference we're talking about here?

Reply:
Here is a good post
http://www.andygup.net/fastest-way-to-find-an-item-in-a-javascript-array/

Reply:
I meant rather in this particular case. How much performance do you expect to gain here by using an object instead of an array?

Reply:
Don't know exactly the value but is more faster than using an array for filtering.

Take in mind we have (for now) four Social Shares added to the attributes, and at least 2 of them have more than a field, to be exactly we have almost 10 fields to update. If we use a `filter`method for updating these particular values, we'll be iterating 10 times per each update.

Reply:
I agree that objects can be more convenient than using `find` or `filter` on an array.

But if you're basing the implementation on performance considerations, it helps to know the approximate amounts of time and memory we're talking about. Are we talking about 10 times faster? Or just 1/1000000 faster? Or even less?

I did a [quick benchmark with a similar scenario](https://jsfiddle.net/68015uek/) and if I do 1000 iterations it's twice as fast using an object. But it's still just 1ms total, and much more iterations than this code is expected to do.

If I run it with 10 iterations, there is no difference at all most of the time.

Reply:
I know that 10 iterations does not affect if you do it in a regular scenario, but we're using react and it's a completely different scenario since aren't ten, are more than ten, since for every change you do on every field it's invoked the `render`method and that is the main performance issue

Reply:
I typed (very slowly) these values, about 70 characters, and for each letter that I type is invoked the `render`method, so, in theory if we implement a `filter`method and taking into account that we have 10 fields that needs to be filtered, we will iterate ~700 times that function. it's a big difference if you compare to an `object`.